securesystemslib/signer/_signer.py

```diff
@@ -225,5 +261,146 @@ def sign(self, payload: bytes) -> Signature:
                 securesystemslib.exceptions.UnsupportedAlgorithmError:
                 Signing errors.
         """
-        sig_dict = sslib_keys.create_signature(self.key_dict, payload)
-        return Signature(**sig_dict)
+        return self._crypto_signer.sign(payload)
+
+
+class CryptoSigner(Signer, metaclass=ABCMeta):
```

Comment:
What's the advantage of deriving here? You say that the individual CryptoSigners could be used as Signers directly but if they don't implement `from_priv_key_uri()` I'm not sure I agree.  The class hierarchy then has a lot of "Signers"  but only some can be used as documented: others will throw a RuntimeError

I think the division of responsibility makes sense but could this just be a clearly internal class that isn't a Signer?